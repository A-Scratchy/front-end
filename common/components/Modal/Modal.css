.ModalCard {
  position: relative;
}

.closeButton {
  top: 1rem;
  right: 1rem;
  background: var(--slate);
  border-radius: 50%;
  border: none;
  position: absolute;
}

.closeButton:hover {
  background-color: var(--slateDark);
  cursor: pointer;
  outline: none;
}

.closeButton:focus {
  outline: 1px solid var(--slate);
  outline-offset: 1px;
}

.closeButtonIcon {
  /* I don't know why, but this is what gets a perfect circle wrapped around the icon */
  fill: var(--mist);
  padding: 0.8rem 0.5rem 0.5rem 0.5rem;
  width: 1rem;
}

.scrollableContainer {
  margin-top: 1rem;
  overflow: auto;
  max-height: 75vh;
}

/* React-Modal Overrides
 * Remove scroll on the body when react-modal is open */
/* stylelint-disable */
:global(.ReactModal__Html--open),
:global(.ReactModal__Body--open) {
  height: 100%;
  overflow: hidden;
}

:global(.ReactModal__Overlay) {
  align-items: center;
  display: flex;
  justify-content: center;
}

:global(.ReactModal__Content) {
  width: 80%;
}
