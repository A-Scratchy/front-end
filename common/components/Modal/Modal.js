import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import ReactModal from 'react-modal';
import ReactGA from 'react-ga';
import CardStyles from 'common/components/Card/Card.css';
import CloseIcon from 'static/images/icons/close_icon.svg';
import ModalStyles from './Modal.css';

Modal.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  hasCloseIcon: PropTypes.bool,
  isOpen: PropTypes.bool,
  onRequestClose: PropTypes.func,
  screenReaderLabel: PropTypes.string.isRequired, // basically a summarizing title
  shouldCloseOnOverlayClick: PropTypes.bool,
};

Modal.defaultProps = {
  className: '',
  hasCloseIcon: true,
  isOpen: false,
  onRequestClose: undefined,
  shouldCloseOnOverlayClick: true,
};

function Modal({
  children,
  className,
  hasCloseIcon,
  isOpen,
  onRequestClose,
  screenReaderLabel,
  shouldCloseOnOverlayClick,
}) {
  if (process.env.NODE_ENV === 'production' && process.env.GOOGLE_ANALYTICS_KEY) {
    ReactGA.modalView(screenReaderLabel);
  }

  return (
    <ReactModal
      className={classNames(CardStyles.Card, ModalStyles.ModalCard, className)}
      contentLabel={screenReaderLabel}
      isOpen={isOpen}
      onRequestClose={onRequestClose}
      shouldCloseOnOverlayClick={shouldCloseOnOverlayClick}
    >
      {hasCloseIcon && (
        <button className={ModalStyles.closeButton} onClick={onRequestClose} type="button">
          <CloseIcon className={ModalStyles.closeButtonIcon} />
        </button>
      )}

      <div className={ModalStyles.scrollableContainer}>{children}</div>
    </ReactModal>
  );
}

export default Modal;
