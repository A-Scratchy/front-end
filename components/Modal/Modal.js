import React from 'react';
import { node, string, bool, func } from 'prop-types';
import classNames from 'classnames';
import ReactModal from 'react-modal';
import { gtag } from 'common/utils/thirdParty/gtag';
import CardStyles from 'components/Cards/Card/Card.module.css';
import CloseButton from 'components/Buttons/CloseButton/CloseButton';
import { SCROLLABLE_CONTAINER } from 'common/constants/testIDs';

import ModalStyles from './Modal.module.css';

Modal.propTypes = {
  children: node.isRequired,
  className: string,
  isOpen: bool,
  onRequestClose: func.isRequired,
  screenReaderLabel: string.isRequired, // basically a summarizing title
  shouldCloseOnOverlayClick: bool,
  isContainerScrollable: bool,
};

Modal.defaultProps = {
  className: undefined,
  isOpen: false,
  shouldCloseOnOverlayClick: true,
  isContainerScrollable: true,
};

function Modal({
  children,
  className,
  isOpen,
  onRequestClose,
  screenReaderLabel,
  shouldCloseOnOverlayClick,
  isContainerScrollable,
}) {
  if (isOpen) {
    gtag.modalView(screenReaderLabel);
  }

  return (
    <ReactModal
      portalClassName={ModalStyles.Modal}
      className={classNames(CardStyles.Card, className)}
      contentLabel={screenReaderLabel}
      isOpen={isOpen}
      onRequestClose={onRequestClose}
      shouldCloseOnOverlayClick={shouldCloseOnOverlayClick}
    >
      <CloseButton onClick={onRequestClose} />
      {isContainerScrollable ? (
        <div className={ModalStyles.scrollableContainer} data-testid={SCROLLABLE_CONTAINER}>
          {children}
        </div>
      ) : (
        children
      )}
    </ReactModal>
  );
}

export default Modal;
