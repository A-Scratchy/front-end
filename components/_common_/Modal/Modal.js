import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import ReactModal from 'react-modal';
import ReactGA from 'react-ga';
import ScreenReaderOnly from 'components/_common_/ScreenReaderOnly/ScreenReaderOnly';
import CardStyles from 'components/_common_/Card/Card.css';
import ModalStyles from './Modal.css';

Modal.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  hasCloseButton: PropTypes.bool,
  isOpen: PropTypes.bool,
  onRequestClose: PropTypes.func.isRequired,
  screenReaderLabel: PropTypes.string.isRequired, // basically a summarizing title
  shouldCloseOnOverlayClick: PropTypes.bool,
};

Modal.defaultProps = {
  className: undefined,
  hasCloseButton: true,
  isOpen: false,
  shouldCloseOnOverlayClick: true,
};

function Modal({
  children,
  className,
  hasCloseButton,
  isOpen,
  onRequestClose,
  screenReaderLabel,
  shouldCloseOnOverlayClick,
}) {
  if (process.env.NODE_ENV === 'production' && process.env.GOOGLE_ANALYTICS_KEY) {
    ReactGA.modalview(screenReaderLabel);
  }

  return (
    <ReactModal
      className={classNames(CardStyles.Card, ModalStyles.ModalCard, className)}
      contentLabel={screenReaderLabel}
      isOpen={isOpen}
      onRequestClose={onRequestClose}
      shouldCloseOnOverlayClick={shouldCloseOnOverlayClick}
    >
      {/* TODO: Style close button */}
      {hasCloseButton && (
        <button onClick={onRequestClose} type="button">
          <ScreenReaderOnly>Close Pop-up</ScreenReaderOnly>X
        </button>
      )}

      <div className={ModalStyles.scrollableContainer}>{children}</div>
    </ReactModal>
  );
}

export default Modal;
