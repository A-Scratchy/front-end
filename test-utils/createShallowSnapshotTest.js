/* eslint-env jest */
import ShallowRenderer from 'react-test-renderer/shallow';

/**
 * Used to create a snapshot test off of surface leaves of the tree.
 * Useful for regression tests on containers or larger components.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const shallowRenderer = new ShallowRenderer();
  const tree = shallowRenderer.render(Component);

  expect(tree).toMatchSnapshot();
};
