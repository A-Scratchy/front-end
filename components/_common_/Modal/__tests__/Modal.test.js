/* eslint-env jest */
import React from 'react';
import { shallow } from 'enzyme';
import ReactGA from 'react-ga';
import createSnapshotTest from 'test-utils/createSnapshotTest';
import createShallowSnapshotTest from 'test-utils/createShallowSnapshotTest';

import Modal from '../Modal';

describe('Modal', () => {
  it('should render with just required props passed', () => {
    createSnapshotTest(
      <Modal screenReaderLabel="Test" onRequestClose={() => {}}>
        Test
      </Modal>,
    );
  });

  it('should render properly with some props assigned and not being open', () => {
    createSnapshotTest(
      <Modal
        className="test-class"
        onRequestClose={() => {}}
        screenReaderLabel="Test"
        shouldCloseOnOverlayClick
      >
        Test
      </Modal>,
    );
  });

  it('should render properly with some props assigned and being open', () => {
    createShallowSnapshotTest(
      <Modal
        className="test-class"
        isOpen
        onRequestClose={() => {}}
        screenReaderLabel="Test"
        shouldCloseOnOverlayClick
      >
        Test
      </Modal>,
    );
  });

  it('should call ReactGA when in prod environment', () => {
    ReactGA.initialize('foo', { testMode: true });

    process.env.NODE_ENV = 'production';
    process.env.GOOGLE_ANALYTICS_KEY = 'debug';

    const props = {
      hasCloseIcon: false,
      onRequestClose: () => {},
      screenReaderLabel: 'Test',
    };

    shallow(<Modal {...props}>Testing</Modal>);

    expect(ReactGA.testModeAPI.calls).toContainEqual(['send', 'pageview', '/modal/Test']);
  });
});
