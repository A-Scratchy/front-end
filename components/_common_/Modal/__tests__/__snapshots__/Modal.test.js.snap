// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render with all props assigned 1`] = `null`;

exports[`Modal should render with required props 1`] = `null`;
